modules/sdk-coin-btc/src/lib/keyPair.ts

```diff
@@ -0,0 +1,96 @@
+import { randomBytes } from 'crypto';
+import * as bip32 from 'bip32';
+import {
+  AddressFormat as BaseAddressFormat,
+  DefaultKeys,
+  isPrivateKey,
+  isPublicKey,
+  isSeed,
+  KeyPairOptions,
+  NotImplementedError,
+  Secp256k1ExtendedKeyPair,
+} from '@bitgo/sdk-core';
+import { address, crypto, networks } from 'bitcoinjs-lib';
+
+const DEFAULT_SEED_SIZE_BYTES = 16;
+
+// TODO: extend the AddressFormat enum in @bitgo/sdk-core with these values
+enum ExtendedAddressFormat {
+  mainnetBech32 = 'mainnetBech32',
+  testnetBech32 = 'testnetBech32',
+}
+
+export type AddressFormatType = BaseAddressFormat | ExtendedAddressFormat;
+export const AddressFormat = { ...BaseAddressFormat, ...ExtendedAddressFormat };
+
+/** Bitcoin key management. */
+export class KeyPair extends Secp256k1ExtendedKeyPair {
+  /**
+   * Public constructor. By default, creates a key pair with a random master seed.
+   *
+   * @param { KeyPairOptions } source Either a master seed, a private key (extended or raw), or a public key
+   *     (extended, compressed, or uncompressed)
+   */
+  constructor(source?: KeyPairOptions) {
+    super(source);
+    if (!source) {
+      const seed = randomBytes(DEFAULT_SEED_SIZE_BYTES);
+      this.hdNode = bip32.fromSeed(seed);
+    } else if (isSeed(source)) {
+      this.hdNode = bip32.fromSeed(source.seed);
+    } else if (isPrivateKey(source)) {
+      this.recordKeysFromPrivateKey(source.prv);
+    } else if (isPublicKey(source)) {
+      this.recordKeysFromPublicKey(source.pub);
+      // TODO: move this to Secp256k1ExtendedKeyPair
+      this.keyPair.compressed = source.pub.length === 66;
+    } else {
+      throw new Error('Invalid key pair options');
+    }
+
+    if (this.hdNode) {
+      this.keyPair = Secp256k1ExtendedKeyPair.toKeyPair(this.hdNode);
+    }
+  }
+
+  /**
+   * Bitcoin uncompressed public and private keys in hex format.
+   *
+   * @returns { DefaultKeys } The keys in the protocol default key format
+   */
+  getKeys(): DefaultKeys {
+    return {
+      pub: this.getPublicKey({ compressed: false }).toString('hex'),
+      prv: this.getPrivateKey()?.toString('hex'),
+    };
+  }
+
+  /**
+   * Get a bitcoin address in legacy or bech32 format.
+   *
+   * @param {AddressFormatType} format One of mainnet, testnet, mainnetBech32, or testnetBech32
+   * @returns {string} The address derived from the public key
```

Comment:
UTXO coins use a triple of keys plus a derivation path to generate address. (I am surprised that we even have an interface to generate addresses controlled by single key)

Deriving addresses from key triples is implemented here

* https://github.com/BitGo/BitGoJS/blob/master/modules/utxo-lib/src/bitgo/wallet/WalletKeys.ts
* https://github.com/BitGo/BitGoJS/blob/master/modules/utxo-lib/src/bitgo/wallet/WalletScripts.ts

Reply:
Do you still need the derivation path if this class is initialized with extended keys? Keep in mind it is using `Secp256k1ExtendedKeyPair` which by default tries to use bip32.
Key deriviation was not contemplated in `Secp256k1ExtendedKeyPair` either but I guess it would make sense to take the derivation path in the constructor parameter so that this class can be initialized with it.
A `derive(...)` method can also be added to the base interface but I guess it would only be useful for a few coins only.

Reply:
Yes, I think so. The motivating idea behind extended key pairs is key derivation. You might as well just generate a non-extended key pair and use that, since that's functionally equivalent to using an extended key pair that only supports the `m` derivation path.